### PR TITLE
fix: change outdir to generate correct code links

### PR DESCRIPTION
**Pull request by john-u**
- set `outDir` to `lib` as it is [required by oclif-dev](https://github.com/oclif/dev-cli/blob/3c44a5bef7d2f2c5c47444648fe2bf34bee39f51/src/commands/readme.ts#L221) to correctly generate readme

relates to SmartThingsCommunity/smartthings-cli#200

## Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **[CONTRIBUTING](../CONTRIBUTING.md)** document
- [x] My code follows the code style of this project (`npm run lint` produces no warnings/errors)
- [x] Any required documentation has been added
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -52,7 +52,7 @@ typings/
 .node_repl_history
 
 # compile dir
-dist
```
Comment (rossiam) [inline .gitignore:55]: Why this particular change? The `lib` convention for transpiled files has always bugged me so when I saw some people used `dist`, I jumped on it.

Comment (john-u) [reply .gitignore:55]: oclif seems opinionated about it being `lib` so `oclif-dev` [assumes it](https://github.com/oclif/dev-cli/blob/3c44a5bef7d2f2c5c47444648fe2bf34bee39f51/src/commands/readme.ts#L221) in some places that break the readme generation.

**Comment by john-u** [comment]
After further investigation, I've found I can achieve the same output by leaving outDir set to `dist` and instead setting the following `repositoryPrefix` in package.json
```
"oclif": {
    "repositoryPrefix": "<%- repo %>/blob/v<%- version %>/<%- _.replace(commandPath, 'dist', 'src') %>"
}
```

However, I am hesitant to patch it this way since it seems every OCLIF CLI I can find open source has `lib` as a Typescript outDir (it's generated this way by the oclif CLI).

**Pull request merged**